python/cuml/benchmark/bench_helper_funcs.py

```diff
@@ -139,13 +147,140 @@ def _build_fil_classifier(m, data, args, tmpdir):
     bst = xgb.train(params, dtrain, num_rounds)
     bst.save_model(model_path)
 
-    return m.load(
-        model_path,
-        algo=args["fil_algo"],
-        output_class=args["output_class"],
-        threshold=args["threshold"],
-        storage_type=args["storage_type"],
-    )
+    fil_kwargs = {}
+    for param, input_name in (
+        ("algo", "fil_algo"),
+        ("output_class", "output_class"),
+        ("threshold", "threshold"),
+        ("storage_type", "storage_type"),
+        ("precision", "precision"),
+    ):
+        try:
+            fil_kwargs[param] = args[input_name]
+        except KeyError:
+            pass
+
+    return m.load(model_path, **fil_kwargs)
+
+
+class OptimizedFilWrapper:
+    """Helper class to make use of optimized parameters in both FIL and
+    experimental FIL through a uniform interface"""
+
+    def __init__(self, fil_model, optimal_chunk_size, experimental):
+        self.fil_model = fil_model
+        self.predict_kwargs = {}
+        if experimental:
+            self.predict_kwargs = {"chunk_size": optimal_chunk_size}
+
+    def predict(self, X):
+        return self.fil_model.predict(X, **self.predict_kwargs)
+
+
+def _build_optimized_fil_classifier(m, data, args, tmpdir):
+    """Setup function for FIL classification benchmarking with optimal
+    parameters"""
+    with using_device_type("gpu"):
+        from cuml.internals.import_utils import has_xgboost
+
+        if has_xgboost():
+            import xgboost as xgb
+        else:
+            raise ImportError("No XGBoost package found")
```

Comment:
Why not just import it at this point? The ImportError raised by line 187 should be self-explanatory, no?

Reply:
Agreed. This should be using our new safe import functionality anyway. Will update accordingly. Not sure why this was done this way in the original.